fs-repo-11-to-12/migration/migration_test.go

```diff
@@ -17,6 +17,24 @@ const (
 	workingRepo = "repotest_copy"
 )
 
+func TestGenericMigration(t *testing.T) {
+	origSetting := EnableFlatFSFastPath
+	defer func() {
+		EnableFlatFSFastPath = origSetting
+	}()
+	EnableFlatFSFastPath = false
+	testMigrationBase(t)
+}
+
+func TestFlatFSMigration(t *testing.T) {
```

Comment:
Unfortunately, running two migration tests results in:

```
    migration_test.go:67: error injecting plugins: already have a datastore named "badgerds"

```

we can either move around the plugin injector code, or just cheat and use unexported global variables here that let the tests work.

Reply:
The datastore plugins end up as singletons anyway (https://github.com/ipfs/go-ipfs/blob/5922eab08f22dc815222bcdc3751165b29c82e28/repo/fsrepo/datastores.go#L67) so I just promoted this to be a package wide singleton